File: NetFramework/Themes/VB/C1CommandBasedApp/C1TrueDBGridOptions.vb
Imports System.Windows.Forms

Imports C1.Win.C1TrueDBGrid

Public Class C1TrueDBGridOptions
	Private _trueGrid As C1TrueDBGrid

	Public Sub New(trueGrid As C1TrueDBGrid)
		_trueGrid = trueGrid
	End Sub

	Public ReadOnly Property TrueGrid() As C1TrueDBGrid
		Get
			Return _trueGrid
		End Get
	End Property

	Public Property DataView() As DataViewEnum
		Get
			Return _trueGrid.DataView
		End Get
		Set
			If DataView = value Then
				Return
			End If

			_trueGrid.DataView = value
			If value = DataViewEnum.GroupBy Then
				If _trueGrid.GroupedColumns.IndexOf(_trueGrid.Columns("Region")) = -1 Then
					_trueGrid.GroupedColumns.Add(_trueGrid.Columns("Region"))
				End If
				For i As Integer = 0 To _trueGrid.RowCount - 1
					_trueGrid.ExpandGroupRow(i)
				Next
			Else
				_trueGrid.GroupedColumns.Clear()
			End If
		End Set
	End Property

	Public Property ShowCaption() As Boolean
		Get
			Return Not String.IsNullOrEmpty(_trueGrid.Caption)
		End Get
		Set
			If ShowCaption = value Then
				Return
			End If

			If value Then
				_trueGrid.Caption = "C1TrueDBGrid caption"
			Else
				_trueGrid.Caption = String.Empty
			End If
		End Set
	End Property

	Public Property ShowFilterBar() As Boolean
		Get
			Return _trueGrid.FilterBar
		End Get
		Set
			_trueGrid.FilterBar = value
		End Set
	End Property

	Public Property ShowGroupByArea() As Boolean
		Get
			Return _trueGrid.GroupByAreaVisible
		End Get
		Set
			_trueGrid.GroupByAreaVisible = value
		End Set
	End Property
End Class

File: NetFramework/Themes/CS/C1RibbonBasedApp/C1TrueDBGridOptions.cs
using System;
using System.Windows.Forms;

using C1.Win.TrueDBGrid;

namespace C1RibbonBasedApp
{
    public class C1TrueDBGridOptions
    {
        private C1TrueDBGrid _trueGrid;

        public C1TrueDBGridOptions(
            C1TrueDBGrid trueGrid)
        {
            _trueGrid = trueGrid;
        }

        public C1TrueDBGrid TrueGrid
        {
            get { return _trueGrid; }
        }

        public DataViewEnum DataView
        {
            get { return _trueGrid.DataView; }
            set
            {
                if (DataView == value)
                    return;

                _trueGrid.DataView = value;
                if (value == DataViewEnum.GroupBy)
                {
                    if (_trueGrid.GroupedColumns.IndexOf(_trueGrid.Columns["Region"]) == -1)
                        _trueGrid.GroupedColumns.Add(_trueGrid.Columns["Region"]);
                    for (int i = 0; i < _trueGrid.RowCount; i++)
                        _trueGrid.ExpandGroupRow(i);
                }
                else
                {
                    _trueGrid.GroupedColumns.Clear();
                }
            }
        }

        public bool ShowCaption
        {
            get { return !string.IsNullOrEmpty(_trueGrid.Caption); }
            set
            {
                if (ShowCaption == value)
                    return;

                if (value)
                    _trueGrid.Caption = "C1TrueDBGrid caption";
                else
                    _trueGrid.Caption = string.Empty;
            }
        }

        public bool ShowFilterBar
        {
            get { return _trueGrid.FilterBar; }
            set { _trueGrid.FilterBar = value; }
        }

        public bool ShowGroupByArea
        {
            get { return _trueGrid.GroupByAreaVisible; }
            set { _trueGrid.GroupByAreaVisible = value; }
        }
    }
}
